internal/stage_index_scan.go

```diff
@@ -56,25 +57,39 @@ func testIndexScan(stageHarness *test_case_harness.TestCaseHarness) error {
 		}
 
 		actualValues := splitBytesToLines(result.Stdout)
-
-		expectedValues, err := getExpectedValuesForQuery(db, testQuery)
-		if err != nil {
-			logger.Errorf("Failed to create test database, this is a CodeCrafters error.")
-			return err
-		}
-
-		if len(actualValues) != len(expectedValues) {
-			return fmt.Errorf("Expected exactly %v lines of output, got: %v", len(expectedValues), len(actualValues))
-		}
-
-		sort.Strings(expectedValues)
-		sort.Strings(actualValues)
-
-		expectedValuesStr := strings.Join(expectedValues, "\n")
-		actualValuesStr := strings.Join(actualValues, "\n")
-
-		if expectedValuesStr != actualValuesStr {
-			return fmt.Errorf("Expected %v to be returned as values, got: %v", expectedValues, actualValues)
+		resultChannel := make(chan []string, 1)
+		errorChannel := make(chan error, 1)
+
+		go func() {
```

Comment:
@andy1li this is lot of complexity for a top-level test function. Let's move it into `getExpectedValuesForQuery`. The same logic should also apply to table_scan, we shouldn't ever see the time exceed 1 second